components/nexus-orient/src/main/java/org/sonatype/nexus/orient/entity/EntityAdapter.java

```diff
@@ -259,33 +295,28 @@ protected void setMetadata(final T entity, final ODocument doc) {
    * Get metadata from entity.
    */
   protected EntityMetadata getMetadata(final T entity) {
-    checkNotNull(entity);
-    EntityMetadata metadata = entity.getEntityMetadata();
-    checkState(metadata != null);
-    return metadata;
+    return metadata(entity);
+  }
+
+  public EntityId encode(final ORID id) {
+    return new EntityId(getRecordIdObfuscator().encode(getType(), id));
+  }
+
+  public ORID decode(final EntityId id) {
+    return getRecordIdObfuscator().decode(getType(), id.toString());
   }
 
   /**
    * Return record identity of entity.
    */
-  protected ORID recordIdentity(final T entity) {
-    EntityMetadata metadata = getMetadata(entity);
-    if (metadata instanceof AttachedEntityMetadata) {
-      return ((AttachedEntityMetadata) metadata).rid;
-    }
-    return getRecordIdObfuscator().decode(getType(), metadata.getId().getValue());
+  public ORID recordIdentity(final T entity) {
+    return recordIdentity(id(entity));
   }
 
-  /**
-   * Return record version of entity.
-   */
-  protected ORecordVersion recordVersion(final T entity) {
-    EntityMetadata metadata = getMetadata(entity);
-    if (metadata instanceof AttachedEntityMetadata) {
-      return ((AttachedEntityMetadata) metadata).rversion;
+  public ORID recordIdentity(final EntityId id) {
+    if (checkNotNull(id) instanceof AttachedEntityId) {
+      return ((AttachedEntityId) id).recordIdentity();
```

Comment:
Would appreciate javadoc explaining the difference in behaviour between `decode` and `recordIdentity` since they have the same signature - otherwise people might use the wrong one expecting it to handle `AttachedEntityId`.